### PR TITLE
Fix foreign keys

### PR DESCRIPTION
**Pull request by Tigrov**
* Fix multiple foreign keys referencing to one table
* Fix possible `null` columns for reference

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

**Review by vjik — COMMENTED** [review]
Should we mark `TableSchema::compositeForeignKey()` as deprecated and remove it in the next major version?

Seems, `TableSchemaInterface::compositeForeignKey()` [used only here](https://github.com/search?q=org%3Ayiisoft%20compositeForeignKey&type=code), and after merge of this PR can be marked as deprecated also.

File: src/Schema.php [hunk 1]
```diff
@@ -210,15 +211,31 @@ protected function loadTableForeignKeys(string $tableName): array
...
+                }
+
+                $fk = (new ForeignKeyConstraint())
+                    ->columnNames(array_column($foreignKey, 'from'))
+                    ->foreignTableName($table)
+                    ->foreignColumnNames(array_column($foreignKey, 'to'))
+                    ->onDelete($foreignKey[0]['on_delete'])
+                    ->onUpdate($foreignKey[0]['on_update']);
+
+                $result[(int) $id] = $fk;
```
Comment (vjik) [inline src/Schema.php:237]:
```suggestion
                $result[] = $fk;
```

`loadTableForeignKeys` returns non-indexed array of foreign keys, right?

**Review by vjik — CHANGES_REQUESTED** [review]
Also need line in changelog

File: src/Schema.php [hunk 2]
```diff
@@ -380,19 +397,17 @@ protected function findColumns(TableSchemaInterface $table): bool
...
 
-            if (!isset($fk[$id])) {
-                $table->foreignKey($id, [$foreignKey['table'], $foreignKey['from'] => $foreignKey['to']]);
-            } else {
-                /** composite FK */
-                $table->compositeForeignKey($id, $foreignKey['from'], $foreignKey['to']);
-            }
+            $foreignReference = array_merge([$foreignKey->getForeignTableName()], $columnNames);
+
+            $table->foreignKey($id, $foreignReference);
```
[the remark this answers is not in the document]
Comment (Tigrov) [reply src/Schema.php:410]: Yes, I tried this way but
> Prior to PHP 8.1, unpacking an array which has a string key is not supported

File: src/Schema.php (hunk 1 above)
Comment (Tigrov) [reply src/Schema.php:237]: Right, looks like need to store `$id` in `ForeignKeyConstraint::$name`

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 ## 1.0.2 under development
 
-- no changes in this release.
+- Bug #268: Fix foreign keys (@Tigrov)
```
Comment (vjik) [inline CHANGELOG.md:5]:
```suggestion
- Bug #268: Fix foreign keys: support multiple foreign keys referencing to one table and possible null columns for reference (@Tigrov)
```

**Review by vjik — APPROVED** [review]

**Review by terabytesoftw — APPROVED** [review]

**Pull request merged**